packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -117,15 +118,22 @@ function initializeHeightsArray<T>({
     const position = positionCache?.get(item);
     if (position) {
       const col = (position.left - centerOffset) / columnWidthAndGutter;
-      // TODO: Change this to handle multi column
-      const isTwoColumnModule =
-        position.width === calculateMultiColumnModuleWidth(columnWidth, gutter, 2);
-      const heightToAdd = position.height + gutter;
-      heights[col] += heightToAdd;
-      if (isTwoColumnModule) {
-        // if position width is greater than columnWidth
-        // increment height of the neighboring column as well
-        heights[col + 1] += heightToAdd;
+      const columnSpan = getItemColumnSpan(item);
+      const moduleHeight = position.height + gutter;
+      if (position.top < heights[col]) {
+        // handles the case where the column height is already taller because of a previously added multi-column module
```

Comment:
this is specifically for the case where we have a multi-column item _before_ a single column

<img width="761" alt="image" src="https://github.com/pinterest/gestalt/assets/6487551/8284ebd5-2d1e-473b-819e-9769d5f30bb9">

in this case, we would skip adding the height for 201 since the column height would already be taller

Reply:
Nice edge case finding!